[Intermittent] UnaryBlockingBasicTestCase.testNonBlockingBallerinaClient fails

Build fails with the following error.
```
[ERROR] Tests run: 413, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 780.553 s <<< FAILURE! - in TestSuite
[ERROR] testNonBlockingBallerinaClient(org.ballerinalang.test.service.grpc.sample.UnaryBlockingBasicTestCase)  Time elapsed: 12.731 s  <<< FAILURE!
java.lang.AssertionError: expected [2] but found [1]
	at org.ballerinalang.test.service.grpc.sample.UnaryBlockingBasicTestCase.testNonBlockingBallerinaClient(UnaryBlockingBasicTestCase.java:150)
```

Refer to https://api.travis-ci.org/v3/job/507716878/log.txt for more info.

Affected version is 0.990.4-SNAPSHOT

--- a/examples/grpc-server-streaming/tests/grpc_server_streaming_test.bal
+++ b/examples/grpc-server-streaming/tests/grpc_server_streaming_test.bal
@@ -21,7 +21,10 @@ function testServerStreamingService() {
     }
 
     int waitCount = 0;
-    while(!completed && (responseMsgs.length() < 3)) {
+    while(true) {
+        if (completed && (responseMsgs.length() == 3)) {
+            break;
+        }
         io:println(responseMsgs);
         runtime:sleep(1000);
         if (waitCount > 10) {

--- a/tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/grpc/sample/UnaryBlockingBasicTestCase.java
+++ b/tests/ballerina-integration-test/src/test/java/org/ballerinalang/test/service/grpc/sample/UnaryBlockingBasicTestCase.java
@@ -141,12 +141,11 @@ public class UnaryBlockingBasicTestCase extends GrpcBaseTest {
     public void testNonBlockingBallerinaClient() {
         Path balFilePath = Paths.get("src", "test", "resources", "grpc", "clients", "unary_nonblocking_client.bal");
         CompileResult result = BCompileUtil.compile(balFilePath.toAbsolutePath().toString());
-        final String serverMsg = "Hello WSO2";
 
         BValue[] responses = BRunUtil.invoke(result, "testUnaryNonBlockingClient", new BValue[]{});
         Assert.assertEquals(responses.length, 1);
-        Assert.assertTrue(responses[0] instanceof BInteger);
-        BInteger responseCount = (BInteger) responses[0];
-        Assert.assertEquals(responseCount.intValue(), 2);
+        Assert.assertTrue(responses[0] instanceof BBoolean);
+        BBoolean response = (BBoolean) responses[0];
+        Assert.assertTrue(response.booleanValue());
     }
 }

--- a/tests/ballerina-integration-test/src/test/resources/grpc/clients/unary_nonblocking_client.bal
+++ b/tests/ballerina-integration-test/src/test/resources/grpc/clients/unary_nonblocking_client.bal
@@ -17,10 +17,11 @@ import ballerina/grpc;
 import ballerina/io;
 import ballerina/runtime;
 
-int total = 0;
 const string ERROR_MSG_FORMAT = "Error from Connector: %s - %s";
+boolean respReceived = false;
+boolean eofReceived = false;
 
-function testUnaryNonBlockingClient() returns int {
+function testUnaryNonBlockingClient() returns boolean {
     // Client endpoint configuration
     HelloWorldClient helloWorldEp = new ("http://localhost:9100");
     // Executing unary non-blocking call registering server message listener.
@@ -28,23 +29,26 @@ function testUnaryNonBlockingClient() returns int {
     if (result is error) {
         string msg = io:sprintf(ERROR_MSG_FORMAT, result.reason(), <string>result.detail().message);
         io:println(msg);
-        return total;
+        return false;
     } else {
         io:println("Connected successfully");
     }
 
     int waitCount = 0;
-    while(total < 2) {
+    while(true) {
+        if (respReceived && eofReceived) {
+            break;
+        }
         runtime:sleep(1000);
-        io:println("msg count: ", total);
+        io:println("response received: ", respReceived);
+        io:println("EOF received: ", eofReceived);
         if (waitCount > 10) {
             break;
         }
         waitCount += 1;
     }
-    io:println("Client got response successfully.");
-    io:println("responses count: " + total);
-    return total;
+    io:println("Response received: " + (respReceived && eofReceived));
+    return (respReceived && eofReceived);
 }
 
 // Server Message Listener.
@@ -53,7 +57,7 @@ service HelloWorldMessageListener = service {
     // Resource registered to receive server messages
     resource function onMessage(string message) {
         io:println("Response received from server: " + message);
-        total = total + 1;
+        respReceived = true;
     }
 
     // Resource registered to receive server error messages
@@ -65,7 +69,7 @@ service HelloWorldMessageListener = service {
     // Resource registered to receive server completed message.
     resource function onComplete() {
         io:println("Server Complete Sending Response.");
-        total = total + 1;
+        eofReceived = true;
     }
 };
 

